maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1331,103 +1335,172 @@ private MavenExecutionRequest populateRequest( CliRequest cliRequest )
         return populateRequest( cliRequest, cliRequest.request );
     }
 
-    @SuppressWarnings( "checkstyle:methodlength" )
     private MavenExecutionRequest populateRequest( CliRequest cliRequest, MavenExecutionRequest request )
     {
+        slf4jLoggerFactory = LoggerFactory.getILoggerFactory();
         CommandLine commandLine = cliRequest.commandLine;
         String workingDirectory = cliRequest.workingDirectory;
         boolean quiet = cliRequest.quiet;
-        boolean showErrors = cliRequest.showErrors;
+        request.setShowErrors( cliRequest.showErrors ); // default: false
+        File baseDirectory = new File( workingDirectory, "" ).getAbsoluteFile();
 
-        String[] deprecatedOptions = { "up", "npu", "cpu", "npr" };
-        for ( String deprecatedOption : deprecatedOptions )
+        handleDeprecatedOptions( commandLine );
+
+        disableOnPresentOption( commandLine, CLIManager.BATCH_MODE, request::setInteractiveMode );
+        enableOnPresentOption( commandLine, CLIManager.SUPRESS_SNAPSHOT_UPDATES, request::setNoSnapshotUpdates );
```

Comment:
The old functionality sets the `noSnapshotUpdates` to false (or true if the option is there).
Because we do not set it now, it would become implicitly false (because it's a primitive boolean). Did you consider setting the default value explicitly?

Reply:
Thanks for the suggestion. I just did that. Also, I rebased against the master branch to be sure the integration test would succeed.